exporter/collector/metrics.go

```diff
@@ -1227,7 +1240,38 @@ func (m *metricMapper) gaugePointToTimeSeries(
 				extraLabels,
 			),
 		},
-	}}
+	})
+
+	// if an untyped prometheus metric from ops agent, double-export as gauge and cumulative
+	// to match GMP exporter behavior. Only for internal use.
+	if untypedDoubleExportFeatureGate.IsEnabled() {
+		val, ok := point.Attributes().Get(GCPOpsAgentUntypedMetricKey)
+		if ok && val.AsString() == "true" {
+			metricKind := metricpb.MetricDescriptor_CUMULATIVE
+			value, valueType := m.numberDataPointToValue(point, metricKind, metric.Unit())
+			series = append(series, &monitoringpb.TimeSeries{
+				Resource:   resource,
+				Unit:       metric.Unit(),
+				MetricKind: metricKind,
+				ValueType:  valueType,
+				Points: []*monitoringpb.Point{{
+					Interval: &monitoringpb.TimeInterval{
+						EndTime: timestamppb.New(point.Timestamp().AsTime()),
+					},
+					Value: value,
+				}},
+				Metric: &metricpb.Metric{
+					Type: t,
```

Comment:
Good catch, it looks like GMP uses a special [`unknown`](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/107933594665ab14ac69f21a66fc6f2c71f130a7/pkg/export/series_cache.go#L326) suffix [when it duplicates the metric](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/cfbb931e82bbefe852a42e8d2eadc5fcf207b38b/pkg/export/series_cache.go#L421-L429). Maybe we should change our metric name function to check for this label

Reply:
Note that it has different suffixes for the gauge portion of the unknown metric (`unknown` and the counter portion of the unknown metric (`unknown:counter`).

Reply:
I think I'm going to refactor this, I think a better approach would be to copy the untyped Gauge pmetric entry into an extra Sum metric, and let the existing path after that handle everything. Then `GetMetricName()` can handle both.

I think using an actual Sum from pdata will also make it easier to handle @ridwanmsharif's question https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/668#discussion_r1261612291. @dashpole wdyt?

Reply:
Works for me, but I'm not convinced the logic should live in the base exporter.  If someone used the prometheus receiver with our base exporter, they would get duplicate timeseries errors.  We need the GMP logic of adding suffixes to prevent duplicate timeseries problems.

Reply:
We can at least move the gate to the GMP exporter so the logic is only active there. Not sure if we could move the logic itself. Maybe through the "extra metrics" approach we did for target and scope info metrics actually

Reply:
I was able to move everything using the extra metrics approach. Ptal, thanks!